### PR TITLE
Enhancement - Deprecate TSLint since it is officially deprecated in favor of ESLin…

### PR DESCRIPTION
**Pull request by jbmatheson**
Deprecated TSLint since it is _officially deprecated_ in favor of ESLint with Typescript rules

Since TSLint has been deprecated. We need to update the linter to use ESlint since it now has official TSLint rules. It is also the recommend linter going forward for Typescript projects.

TESTING STEPS:
1. After running `npm install` in UI.Frontend module, run `npm run dev`.
2. There should be one warning that will output. It is related to the unused var assigned in `HelloWorld.ts`.

This shows the new linter is set up and configured correctly. 

## Motivation and Context

This fixes the inevitable problem of TSLint being deprecated and, as such, no longer receiving active support. It also aligns our linter with the new standard, which is ESLint with Typescript definitions

## How Has This Been Tested?

Yes, see above steps for testing. This can be tested with either the `dev` or `prod` flag.

## Types of changes

Deprecation replacement.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by godanny86** [comment]
👍 to this

File: src/main/archetype/ui.frontend/package.json [hunk 1]
```diff
@@ -1,56 +1,60 @@
...
+    "optimize-css-assets-webpack-plugin": "^5.0.1",
+    "postcss-loader": "^3.0.0",
+    "sass": "^1.17.2",
+    "sass-loader": "^7.1.0",
+    "source-map-loader": "^0.2.4",
+    "style-loader": "^0.14.1",
+    "terser-webpack-plugin": "^1.4.1",
+    "ts-loader": "^5.3.3",
+    "tsconfig-paths-webpack-plugin": "^3.2.0",
+    "tslint": "^5.13.0",
```
Comment (richardhand) [inline src/main/archetype/ui.frontend/package.json]: @jbmatheson - can the tslint dev dependencies be removed?

Comment (jbmatheson) [reply src/main/archetype/ui.frontend/package.json]: @richardhand - Good catch. I have now removed `tslint1` `tslint-loader` and `tslint-webpack-plugin` as they are no longer needed.

**Comment by jbmatheson** [comment]
@richardhand - PR updated to HEAD

**Pull request merged**